crates/libsyntax2/src/lib.rs

```diff
@@ -144,6 +144,21 @@ fn find_reparsable_node(node: SyntaxNodeRef, range: TextRange) -> Option<(Syntax
         let res = match node.kind() {
             BLOCK => grammar::block,
             NAMED_FIELD_DEF_LIST => grammar::named_field_def_list,
+            NAMED_FIELD_LIST => grammar::named_field_list,
+            ENUM_VARIANT_LIST => grammar::enum_variant_list,
+            MATCH_ARM_LIST => grammar::match_arm_list,
+            USE_TREE_LIST => grammar::use_tree_list,
+            EXTERN_ITEM_LIST => grammar::extern_item_list,
+            TOKEN_TREE => grammar::token_tree,
```

Comment:
I think this is slightly wrong. Token tree might use `() or `[]`, and not `{}`. We maintain `{}` structure carefully, but that's not true for `()` and `[]`.  I think we should make an additional check here that first child is `{`.

The test for it probbaly would be like

```
fn foo() {
    vec![<|><|>]
}
```

current implementation I think will try to reparse `[]` and will faile the `is_balanced` check. The correct impl should reparse the fn body.